packages/core-data/src/entities.js

```diff
@@ -400,9 +400,12 @@ export const getOrLoadEntitiesConfig =
 	async ( { select, dispatch } ) => {
 		let configs = select.getEntitiesConfig( kind );
 		if ( configs && configs.length !== 0 ) {
-			if ( window.__experimentalEnableSync ) {
-				registerSyncConfigs( configs );
+			if ( process.env.IS_GUTENBERG_PLUGIN ) {
+				if ( window.__experimentalEnableSync ) {
```

Comment:
Could these two checks be combined? (same for the other one)

Reply:
They can as long as we add `// eslint-disable-next-line @wordpress/is-gutenberg-plugin` above (the lint rule doesn't seem to recognise if statements with multiple conditions)

Reply:
ohhh, I think that rule is actually important to follow because that's what allows the tree shaking, so I actually we shouldn't do that (even in the case where it's done now in the PR)

Reply:
Oooh I see. For the places where I disabled the rule, I guess we could wrap the whole block in the condition, but then we'd have to duplicate some of the logic to make sure the `else` part underneath still runs if we're in the plugin but have the experiment disabled. @mikachan are you able to take over? It's dinner time for me 😅

Reply:
... or would it work if we added the `IS_GUTENBERG_PLUGIN` check inside the `window.__experimentalEnableSync` one?

Reply:
> ... or would it work if we added the IS_GUTENBERG_PLUGIN check inside the window.__experimentalEnableSync one?

I've just tried this and the linter seemed happy 👀

Reply:
@youknowriad, do you think the latest change that moves the `IS_GUTENBERG_PLUGIN` checks is good to merge?